wagtail/contrib/forms/models.py

```diff
@@ -115,6 +119,61 @@ def clean_name(self):
         FieldPanel('default_value', classname="formbuilder-default"),
     ]
 
+
+    def save(self, *args, **kwargs):
+        """
+        When new fields are created, generate a template safe ascii name to use as the
+        JSON storage reference for this field. Previously created fields will be updated
+        to use the legacy unidecode method via checks & _migrate_legacy_clean_name.
+        """
+
+        is_new = self.pk is None
+        if is_new:
+            clean_name = get_field_clean_name(self.label)
+            self.clean_name = clean_name
+
+        super().save(*args, **kwargs)
+
+
+    @classmethod
+    def _migrate_legacy_clean_name(cls):
+        """
+        Ensure that existing data stored will be accessible via the legacy clean_name.
+        When checks run, replace any blank clean_name values with the unidecode conversion.
+        """
+
+        try:
+            objects = cls.objects.filter(clean_name__exact='')
+
+            for obj in objects:
+                legacy_clean_name = str(slugify(string_to_ascii(obj.label)))
+                obj.clean_name = legacy_clean_name
+                obj.save()
+
+            if objects.count() > 0:
+                return Info(
+                    'Added `clean_name` on %s form field(s)' % objects.count(),
+                    obj=cls
+                )
+
+        except FieldError:
```

Comment:
Oops, looks like that was bad advice - on an unmigrated bakerydemo, I'm now getting

```
django.db.utils.ProgrammingError: column base_formfield.clean_name does not exist
LINE 1: ...COUNT(*) AS "__count" FROM "base_formfield" WHERE "base_form...
```
presumably because the database query (and thus the error) only happens on the `objects.count()` line below. Will fix...

Reply:
Fixed in b8c36c0d7df23f19f3be61e28246f32484c26644